### DISCUSSION TITLE
Invert Image inverts the alpha channel

### DISCUSSION THREAD

**Discussion by mnpenner** [Comfy-Org/ComfyUI#5619] [General]
### Expected Behavior

![image](https://github.com/user-attachments/assets/865f454c-5d1c-4dd0-9743-a40044f13929)

### Actual Behavior

![image](https://github.com/user-attachments/assets/4dd32ab6-0d88-46c9-92e8-5f56e41f09f3)

### Steps to Reproduce

Invert an image with an alpha channel

### Debug Logs

```powershell
N/A
```

### Other

https://github.com/comfyanonymous/ComfyUI/blob/3b9a6cf2b11094f92228b121c6a0d466ba5d5246/nodes.py#L1728-L1741

Here's the corrected image inversion w/ the help of ChatGPT:

```py
def invert_image(tensor: torch.Tensor):
    """
    Invert RGB values in a tensor image, preserving the alpha channel if present.
    Handles both [B, H, W, C] and [H, W, C] shapes.

    Args:
        tensor (torch.Tensor): Input tensor image. Shape: [B, H, W, C] or [H, W, C].

    Returns:
        torch.Tensor: Inverted tensor image.
    """
    # Ensure tensor has at least 4 dimensions (batch dimension)
    is_single_image = len(tensor.shape) == 3
    if is_single_image:
        # Add batch dimension: [H, W, C] -> [1, H, W, C]
        tensor = tensor.unsqueeze(0)

    # Separate RGB and Alpha channels
    rgb = tensor[..., :3]  # First three channels (R, G, B)
    # Alpha channel if present
    alpha = tensor[..., 3:] if tensor.shape[-1] > 3 else None

    # Invert RGB channels
    inverted_rgb = 1 - rgb

    # Combine inverted RGB with original Alpha channel if present
    if alpha is not None:
        inverted_tensor = torch.cat((inverted_rgb, alpha), dim=-1)
    else:
        inverted_tensor = inverted_rgb

    # Remove batch dimension if input was a single image
    if is_single_image:
        inverted_tensor = inverted_tensor.squeeze(0)

    return inverted_tensor
```

**Comment by ltdrdata** [comment]
Originally, ComfyUI handles alpha channels separately as masks and doesn't deal with them directly.

However, since features handling alpha channels have been increasing recently, it might be good to consider this aspect as well.

I'm not sure if it's the right approach to avoid inverting the alpha channel unconditionally.

I think this needs discussion.
Personally, I think implementing 'Invert Image' and 'Invert Image (alpha)' as separate nodes would be appropriate.

**Reply by mnpenner** [reply]
Would 'Invert Image (alpha)'  invert the color channels and leave the alpha channel, as depicted? Or just invert the alpha channel?

Separate nodes sounds fine to me. It'd be safer/backwards-compatible that way.
